src/Runner.Common/JobServer.cs

```diff
@@ -40,19 +41,33 @@ public async Task ConnectAsync(VssConnection jobConnection)
             {
                 try
                 {
+                    try
+                    {
+                        Trace.Info("Requesting Actions health endpoint status");
+                        // Construct the HttpClient object directly, without going through GetClient<T>
+                        // as we haven't properly connected to the server at this point
+                        var client = new HealthHttpClient(_connection.Uri, _connection.InnerHandler, false);
```

Comment:
I would move this into the exception catch.
If we fail to call `ConnectAsync()`, let's try the health endpoint.
Also, can we make a request to github.com as well?